### PR TITLE
fix(pg) add postgress kb annotation and comments

### LINKED ISSUE #552: Should apply kb annotation and comments to the PostgresSQLSpec
So far, missing kb annotation as well as comments.

```go

type PostgresSQLSpec struct {
	harbormetav1.ImageSpec `json:",inline"`

	Storage          string                      `json:"storage,omitempty"`
	Replicas         int                         `json:"replicas,omitempty"`
	StorageClassName string                      `json:"storageClassName,omitempty"`
	Resources        corev1.ResourceRequirements `json:"resources,omitempty"`
	SslConfig        string                      `json:"sslConfig,omitempty"`
	ConnectTimeout   int                         `json:"connectTimeout,omitempty"`
}

```

### PR DESCRIPTION
**Pull request by Ghostbaby**
add postgress kb annotation and comments

fixed #552

Signed-off-by: zhuhuijun <zhuhuijunzhj@gmail.com>

### CODE REVIEW

**Comment by bitsf** [comment]
should regenerate files if update types

File: apis/goharbor.io/v1alpha3/harborcluster_types.go [hunk 1]
```diff
@@ -95,12 +95,19 @@ type Database struct {
 type PostgresSQLSpec struct {
 	harbormetav1.ImageSpec `json:",inline"`
 
-	Storage          string                      `json:"storage,omitempty"`
-	Replicas         int                         `json:"replicas,omitempty"`
-	StorageClassName string                      `json:"storageClassName,omitempty"`
-	Resources        corev1.ResourceRequirements `json:"resources,omitempty"`
-	SslConfig        string                      `json:"sslConfig,omitempty"`
-	ConnectTimeout   int                         `json:"connectTimeout,omitempty"`
+	// Storage is define database data store pvc size
```
Comment (steven-zou) [inline apis/goharbor.io/v1alpha3/harborcluster_types.go:98]:
```suggestion
	// Storage defines database data store pvc size
```

File: apis/goharbor.io/v1alpha3/harborcluster_types.go [hunk 2]
```diff
@@ -95,12 +95,19 @@ type Database struct {
...
+	// +kubebuilder:validation:Optional
+	Storage string `json:"storage,omitempty"`
+	// Replicas is define database instance replicas
```
Comment (steven-zou) [inline apis/goharbor.io/v1alpha3/harborcluster_types.go:101]:
```suggestion
	// Replicas defines database instance replicas
```

File: apis/goharbor.io/v1alpha3/harborcluster_types.go [hunk 3]
```diff
@@ -95,12 +95,19 @@ type Database struct {
...
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum:=1
+	Replicas int `json:"replicas,omitempty"`
+	// StorageClassName is define use which StorageClass to create pvc
```
Comment (steven-zou) [inline apis/goharbor.io/v1alpha3/harborcluster_types.go:105]:
```suggestion
	// StorageClassName defines use which StorageClass to create pvc
```

File: apis/goharbor.io/v1alpha3/harborcluster_types.go [hunk 4]
```diff
@@ -95,12 +95,19 @@ type Database struct {
...
+	// +kubebuilder:validation:Optional
+	StorageClassName string `json:"storageClassName,omitempty"`
+	// Resources is define database pod resource config
```
Comment (steven-zou) [inline apis/goharbor.io/v1alpha3/harborcluster_types.go:108]:
```suggestion
	// Resources defines database pod resource config
```

File: apis/goharbor.io/v1alpha3/harborcluster_types.go [hunk 5]
```diff
@@ -95,12 +95,19 @@ type Database struct {
...
+	// +kubebuilder:validation:Optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
```
Comment (steven-zou) [inline apis/goharbor.io/v1alpha3/harborcluster_types.go:110]: As all the fields are optional, do the default values are correctly handled?

**Review by steven-zou — CHANGES_REQUESTED** [review]
Please do changes.

File: apis/goharbor.io/v1alpha3/harborcluster_types.go (hunk 5 above)
Comment (Ghostbaby) [reply apis/goharbor.io/v1alpha3/harborcluster_types.go:110]: yes, postgress-operator will set the default value.

**Review by steven-zou — APPROVED** [review]
LGTM

**Review by heww — APPROVED** [review]

**Pull request merged**